ios/UnityAds/UnityAdsCampaign/UnityAdsCampaignManager.m

```diff
@@ -137,6 +137,28 @@ - (void)_processCampaignDownloadData {
       NSString *gamerId = [jsonDictionary objectForKey:kUnityAdsGamerIDKey];
       
       [[UnityAdsProperties sharedInstance] setGamerId:gamerId];
+
+      NSError *error;
+      NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:[[UnityAdsCacheManager sharedInstance] getCachePath] error:&error];
+      UALOG_DEBUG(@"Contents of path: %@", files);
+
+      [files enumerateObjectsUsingBlock:^(NSString *file, NSUInteger idx, BOOL *stop) {
+        UALOG_DEBUG(@"Current file: %@", file);
+        if ([self getCampaignWithVideoFile:file] == nil) {
+          UALOG_DEBUG(@"Should delete this file, not in campaigns: %@", [[[UnityAdsCacheManager sharedInstance] getCachePath] stringByAppendingPathComponent:file]);
+          NSError *error;
+          BOOL success = [[NSFileManager defaultManager] removeItemAtPath:[[[UnityAdsCacheManager sharedInstance] getCachePath] stringByAppendingPathComponent:file] error:&error];
+          if (!success) {
+            UALOG_DEBUG(@"Could not remove item at path: %@", [[[UnityAdsCacheManager sharedInstance] getCachePath] stringByAppendingPathComponent:file]);
+          }
+          else {
+            UALOG_DEBUG(@"Success removing item at path: %@", [[[UnityAdsCacheManager sharedInstance] getCachePath] stringByAppendingPathComponent:file]);
```

Comment:
Move [[[UnityAdsCacheManager sharedInstance] getCachePath] to a temp var instead of calling it multiple times.